pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
we should not delete the cluster when reconcile fails. we should keep the trp so that user can see why it fails. user should delete the tpr themselves.